pkg/controller/garbagecollector/operations.go

```diff
@@ -112,9 +103,18 @@ func (gc *GarbageCollector) removeFinalizer(owner *node, targetFinalizer string)
 			klog.V(5).Infof("the %s finalizer is already removed from object %s", targetFinalizer, owner.identity)
 			return nil
 		}
+
 		// remove the owner from dependent's OwnerReferences
-		ownerObject.SetFinalizers(newFinalizers)
-		_, err = gc.updateObject(owner.identity, ownerObject)
+		patch, err := json.Marshal(map[string]interface{}{
```

Comment:
I don't think it makes any difference, but we do have a struct defined for this purpose: https://github.com/kubernetes/kubernetes/blob/13cde9e18115cd8b1e6c8e328be8666bf1a7437b/pkg/controller/garbagecollector/patch.go#L75-L82

Reply:
Will do it when I introduce the partial object client